tests/python/pants_test/test_base.py

```diff
@@ -339,6 +362,8 @@ def _init_engine(cls):
       native=init_native(),
       build_configuration=cls.build_config(),
       build_ignore_patterns=None,
+      # Stu assures me that this isn't necessary, but sources_for doesn't work without it, so...
```

Comment:
This is actually precisely the case where it might be necessary... you're providing it as a parameter (ie, a `root` in the graph).

But yea, hopefully this will be clearer after #5788.